packages/server/lib/cross-origin/process-callback.ts

```diff
@@ -0,0 +1,67 @@
+import { getFullWebpackOptions } from '@cypress/webpack-batteries-included-preprocessor'
+import md5 from 'md5'
+import { fs } from 'memfs'
+import * as path from 'path'
+import webpack from 'webpack'
+
+const VirtualModulesPlugin = require('webpack-virtual-modules')
+
+interface Options {
+  file: string
+  fn: string
+}
+
+// @ts-ignore - webpack expects `fs.join` to exist for some reason
+fs.join = path.join
+
+export const processCallback = ({ file, fn }: Options) => {
+  const source = fn.replace(/Cypress\.require/g, 'require')
+  const webpackOptions = getFullWebpackOptions(file, require.resolve('typescript'))
+
+  const inputFileName = md5(source)
+  const inputDir = path.dirname(file)
+  const inputPath = path.join(inputDir, inputFileName)
+  const outputDir = '/'
+  const outputFileName = 'output'
+  const outputPath = `${outputDir}${outputFileName}.js`
+
+  const modifiedWebpackOptions = {
+    ...webpackOptions,
+    entry: {
+      [outputFileName]: inputPath,
+    },
+    output: {
+      path: outputDir,
+    },
+    plugins: [
+      new VirtualModulesPlugin({
```

Comment:
what does this plugin give us?

Reply:
It allows the entry to be an in-memory source code string instead of a file on disk.

Reply:
So the compiled output of these callbacks is then cleaned up when the server process exits?

Reply:
Since it's all in-memory, it should get garbage-collected earlier than that since it's no longer needed after the webpack compilation is finished.